src/unstract/sdk/index.py

```diff
@@ -236,13 +236,17 @@ def index_file(
         full_text = []
         extracted_text = ""
         try:
-            x2text = X2Text(tool=self.tool)
-            x2text_adapter_inst: X2TextAdapter = x2text.get_x2text(
-                adapter_instance_id=x2text_adapter
-            )
-            extracted_text = x2text_adapter_inst.process(
-                input_file_path=file_path, output_file_path=output_file_path
-            )
+            if not output_file_path:
```

Comment:
Based on what does the caller understand whether output_file_path needs to be passed or not? @Deepak-Kesavan

Reply:
@gaya3-zipstack Currently we use `output_file_path`, to store the extracted text. If the `output_file_path` is not present, the indexing is for `summarize` and the file is already a text.

`output_file_path` is optional and it is added in the doc string of the `index_file` function

Reply:
@Deepak-Kesavan 
> If the output_file_path is not present, the indexing is for summarize and the file is already a text.

This won't be immediately obvious to a developer, I would advice against implicitly performing such logic. Anyway this function is due for a refactor